File: src/main/java/com/baulsupp/oksocial/network/DnsMode.java
package com.baulsupp.oksocial.network;

import com.baulsupp.oksocial.output.util.UsageException;

public enum DnsMode {
  JAVA,
  NETTY,
  DNSGOOGLE;

  public static DnsMode fromString(String dnsMode) {
    switch (dnsMode) {
      case "java":
        return DnsMode.JAVA;
      case "netty":
        return DnsMode.NETTY;
      case "dnsgoogle":
        return DnsMode.DNSGOOGLE;
      default:
        throw new UsageException("unknown dns mode '" + dnsMode + "'");
    }
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/network/DnsMode.kt
package com.baulsupp.oksocial.network

import com.baulsupp.oksocial.output.util.UsageException

enum class DnsMode {
  JAVA,
  NETTY,
  GOOGLE;

  companion object {
    @JvmStatic
    fun fromString(dnsMode: String): DnsMode =
      DnsMode.values().find { it.name.toLowerCase() == dnsMode } ?: throw UsageException(
        "unknown dns mode '$dnsMode'")
  }
}
